nilearn/decoding/space_net.py

```diff
@@ -812,20 +812,8 @@ def _check_params(self):
                 f"{self.SUPPORTED_PENALTIES}. "
                 f"Got {self.penalty}."
             )
-        if self.loss is not None and self.loss not in self.SUPPORTED_LOSSES:
-            raise ValueError(
-                f"'loss' parameter must be one of {self.SUPPORTED_LOSSES}. "
-                f"Got {self.loss}."
-            )
-        if (
-            self.loss is not None
-            and not self.is_classif
-            and (self.loss == "logistic")
-        ):
-            raise ValueError(
-                "'logistic' loss is only available for classification "
-                "problems."
-            )
+        if self._is_classification:
+            self._validate_loss(self.loss)
```

Comment:
The base class would not know which classes will extend it. So it should not make such checks depending on the type of the inheriting class. The inheriting class should overwrite `_check_params` function, call super class `_check_params` for common checks and continue for its own checks. For example in `SpaceNetClassifier` we should have :

```python
def _check_params(self):
    super()._check_params()
    self._validate_loss(self.loss)
```